cli/ops/plugin.rs

```diff
@@ -151,3 +165,59 @@ impl Drop for PluginOpAsyncFuture {
     self.fut.take();
   }
 }
+
+struct PluginResourceTable<'a> {
+  inner: RefMut<'a, ResourceTable>,
+  plugin_lib: &'a Rc<Library>,
+}
+
+impl<'a> PluginResourceTable<'a> {
+  fn new(
+    resource_table: &'a mut Rc<RefCell<ResourceTable>>,
+    plugin_lib: &'a Rc<Library>,
+  ) -> Self {
+    Self {
+      inner: resource_table.borrow_mut(),
+      plugin_lib,
+    }
+  }
+}
+
+/// A resource that has been inserted in the resource table by a dynamically
+/// loaded plugin.
+struct RefResource {
+  inner: Box<dyn Resource>,
+  _plugin_lib: Rc<Library>,
+}
+
+impl<'a> plugin_api::ResourceTable for PluginResourceTable<'a> {
+  fn add(&mut self, name: &str, resource: Box<dyn Resource>) -> ResourceId {
+    // Resources inserted by plugins are wrapped in a RefResource wrapper, so
+    // we can ensure a reference to Rc<Library> is kept for as long as the
+    // resource stays in the resource table.
```

Comment:
Isn't this condition implied by `PluginResourceTable` storing `plugin_lib`?

Reply:
No if you close the plugin resource then resources created by the op can still remain in the table.

Reply:
Oh i see what you mean. The PluginResourceTable object is really short lived -- it's basically a wrapper around RefMut<ResourceTable>. It gets dropped when an op returns.

Reply:
Then when are they closed? This is surprising, I would expect that closing plugin resource results in immediate drop of its associated data.

Reply:
> Oh i see what you mean. The PluginResourceTable object is really short lived -- it's basically a wrapper around `RefMut<ResourceTable>` -- it gets dropped when an op dispatcher returns.

I see, can you add a comment about that? It's really not obvious at first glance

Reply:
> Then when are they closed? This is surprising, I would expect that closing plugin resource results in immediate drop of its associated data.

Yeah, but how would that work? When the 'Drop' handler for 'PluginResource' runs it *could * loop over the resource table, but it currently has no way to identify which resources were created by the dropped resource.

Reply:
> I see, can you add a comment about that? It's really not obvious at first glance

Sure. Or would renaming it to `ResourceTableRef` help?

Reply:
Did the rename, added the comment as requested.